### PR TITLE
API: Default ExtensionArray.astype

### PR DESCRIPTION
**Pull request by TomAugspurger**
I need this to progress cleanly on https://github.com/pandas-dev/pandas/pull/19558 and (which is blocking https://github.com/pandas-dev/pandas/pull/19520).

### CODE REVIEW

File: pandas/core/arrays/base.py [hunk 1]
```diff
@@ -138,6 +140,34 @@ def nbytes(self):
...
+            new dtype.
+
+        Returns
+        -------
+        array : ndarray
+            NumPy ndarray with 'dtype' for its dtype.
+        """
+        np_dtype = np.dtype(dtype)
+
+        if np_dtype != 'object':
```
Comment (jreback) [inline pandas/core/arrays/base.py]: use is_object_dtype

Comment (jreback) [reply pandas/core/arrays/base.py]: why do you even need this?

File: pandas/tests/extension_arrays/test_common.py [hunk 2]
```diff
@@ -0,0 +1,36 @@
+import numpy as np
```
Comment (jreback) [inline pandas/tests/extension_arrays/test_common.py]: put this in ``pandas/tests/extension/test_common.py`` you also need an ``__init__.py``

File: pandas/tests/extension_arrays/test_common.py [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+
+import pandas.util.testing as tm
+from pandas.core.arrays import ExtensionArray
```
Comment (jreback) [inline pandas/tests/extension_arrays/test_common.py]: would also move the existing extension tests here

**Review by jreback — CHANGES_REQUESTED** [review]

File: pandas/core/arrays/base.py (hunk 1 above)
Comment (TomAugspurger) [reply pandas/core/arrays/base.py]: You mean the check for object? Fair point. If the underlying object supports conversion to whatever format, we should allow it.

```python
In [1]: import pandas_ip as ip

In [2]: ip.IPAddress([1, 2, 3]).astype(object)
Out[2]:
array([IPv4Address('0.0.0.1'), IPv4Address('0.0.0.2'),
       IPv4Address('0.0.0.3')], dtype=object)

In [3]: ip.IPAddress([1, 2, 3]).astype(int)
Out[3]: array([1, 2, 3])

```

Which simplifies things nicely!

**Comment by TomAugspurger** [comment]
cc @jorisvandenbossche if you have thoughts.

**Review by jorisvandenbossche — APPROVED** [review]

**Pull request merged**